packages/hebao_v1/contracts/base/MetaTxModule.sol

```diff
@@ -88,19 +88,25 @@ contract MetaTxModule is BaseModule
     /// @param gasToken The token to pay the relayer as fees. Use address(0) for Ether.
     /// @param gasPrice The amount of `gasToken` to pay per gas. 0 is a valid value.
     /// @param gasLimit The max amount of gas that can be used by this meta transaction,
-    ///                excluding an extra `GAS_OVERHEAD` for paying the relayer fees.
+    ///                including the `gasReserve`.
+    /// @param gasReserve The amount of `gasToken` reserved for the meta data storage and
+    ///                   extra logic in the meta-transaction.
     /// @param signatures Signatures.
     function executeMetaTx(
         bytes   calldata data,
         uint    nonce,
         address gasToken,
         uint    gasPrice,
         uint    gasLimit,
+        uint    gasReserve,
         bytes[] calldata signatures
         )
         external
         payable
     {
+        require(gasReserve >= MIN_GAS_RESERVE, "INVALID_GAS_RESERVE");
+        require(gasLimit > gasReserve, "INVALID_GAS_LIMIT");
```

Comment:
I don't think we need to do this check. I would just charge `gasReserve` on top of `gasSpent` without any other rules (`gasReserve` could be a lot higher than the gas needed to actually execute the code on-chain). So just using `gasLimit` as the actual gas limit on-chain seems easier.